pip/commands/install.py

```diff
@@ -224,6 +236,14 @@ def run(self, options, args):
         if options.build_dir:
             options.build_dir = os.path.abspath(options.build_dir)
 
+        allowed_strategies = ["eager", "non-eager"]
```

Comment:
You could instead just set add ``choices=["eager", "non-eager"]`` to the ``add_option`` call above. It should handle the error handling for you.